src/main/java/uk/antiperson/stackmob/config/MainConfig.java

```diff
@@ -1,56 +1,218 @@
+    private final List<EntityType> stack_threshold_enabled = new ArrayList<>();
+    private final Map<EntityType, Integer> stack_threshold = new HashMap<>();
+
+    private final Map<EntityType, String> tag_format = new HashMap<>();
+    private final Map<EntityType, Integer> tag_threshold = new HashMap<>();
+    private final Map<EntityType, TagMode> tag_tag_mode = new HashMap<>();
+    private Integer[] tag_nearby_radius;
+
+    private final List<EntityType> drop_multi = new ArrayList<>();
+    private final List<EntityType> drop_loot_table = new ArrayList<>();
+    private final List<EntityType> drop_type_blacklist = new ArrayList<>();
+    private final Map<EntityType, List<EntityDamageEvent.DamageCause>> drop_reason_blacklist = new HashMap<>();
+    private final Map<EntityType, List<Material>> drop_item_blacklist = new HashMap<>();
+    private final Map<EntityType, List<Material>> drop_item_one_per = new HashMap<>();
+
+    private final List<EntityType> exp_multi = new ArrayList<>();
+    private final List<EntityType> exp_type_blacklist = new ArrayList<>();
+    private final Map<EntityType, Double> exp_mult_min = new HashMap<>();
+    private final Map<EntityType, Double> exp_mult_max = new HashMap<>();
+
+    private final List<EntityType> player_stats = new ArrayList<>();
+
+    private final List<EntityType> waiting_enabled = new ArrayList<>();
+    private final List<EntityType> waiting_type_blacklist = new ArrayList<>();
+    private final Map<EntityType, List<CreatureSpawnEvent.SpawnReason>> waiting_reason_blacklist = new HashMap<>();
+    private final Map<EntityType, Integer> waiting_time = new HashMap<>();
+
+    private final Map<EntityType, Integer> death_min_step = new HashMap<>();
+    private final Map<EntityType, Integer> death_max_step = new HashMap<>();
+    private final Map<EntityType, Map<String, List<EntityType>>> death_type_blacklist = new HashMap<>();
+    private final Map<EntityType, Map<String, List<EntityDamageEvent.DamageCause>>> death_reason_blacklist = new HashMap<>();
+    private final Map<EntityType, Collection<String>> death_priority = new HashMap<>();
+
+    private final List<EntityType> targeting_disabled = new ArrayList<>();
+    private final List<EntityType> targeting_type_blacklist = new ArrayList<>();
+    private final Map<EntityType, List<EntityTargetEvent.TargetReason>> targeting_reason_blacklist = new HashMap<>();
+
+    private final List<EntityType> entity_type_blacklist = new ArrayList<>();
+    private final Map<EntityType, List<CreatureSpawnEvent.SpawnReason>> entity_reason_blacklist = new HashMap<>();
+    private final Map<EntityType, List<World>> entity_world_blacklist = new HashMap<>();
+
+    private final Map<EntityType, Map<String, Boolean>> remove_stack_data = new HashMap<>();
+    private final Map<EntityType, Map<String, ListenerMode>> listener_mode = new HashMap<>();
+
+    private boolean multi_slime;
+
```

Comment:
Is the increased memory usage warranted here? Config values are already loaded into memory by bukkit, so the performance difference would be negligible?

Reply:
Indeed, gets for listener mode and milti_slime options are rare but there is a difference, "almost negligible". On a populated server, I think theses options may have an impact.

Reply:
*The rest is important in reality, because bukkit config is a bit heavier, even in memory.

Reply:
For instance, I'm making a two spark repors (one before, one after) this evening ^^

Reply:
With same circumstances, ignore blank lines, focus on highlighted ones :

- With config caching ->
![image](https://user-images.githubusercontent.com/43724816/85229429-d9bcc380-b3e9-11ea-9924-05686009598c.png)



- Without -> 
![image](https://user-images.githubusercontent.com/43724816/85229412-c4479980-b3e9-11ea-822f-133f1aded75c.png)

On the first report, you also view the other optimizations. They are more remarkable with lots of mob spawning at "the same place", for example with mob spawners.

Concerning memory usage, it's indeed bigger, but I didn't notice any weird or abnormal usage.

Without : https://spark.lucko.me/#z7ndThxHCN
With : https://spark.lucko.me/#IZDv8V82FS

Reply:
I think usage could be reduced for Lists by "inverting them", only storing disabled for example or enabled according to the config option.